CRM/Sumfields/Form/SumFields.php

```diff
@@ -168,7 +181,10 @@ function postProcess() {
     $session = CRM_Core_Session::singleton();
 
     sumfields_save_setting('generate_schema_and_data', 'scheduled:'. date('Y-m-d H:i:s'));
-    if ($values['when_to_apply_change'] == 'on_submit') {
+    // Save our form page settings
+    sumfields_save_setting('data_update_method', $values['data_update_method']);
+
```

Comment:
Also here - we definitely need to save the new setting. But do we need to change the message to the user? If the user chooses to save settings on submit then the changes will be saved AND the data will be generated immediately - regardless of whether or not they have chosen to update data via triggers.

Reply:
Then I suppose I should include the setting with sumfields_save_setting . Also on the sumfields.settings.php  and lastly, I should refine (or even revert) the select cases there, correct?